zvariant/src/lib.rs

```diff
@@ -1,3 +1,4 @@
+#![recursion_limit = "256"]
```

Comment:
It's a crate-level attribute. Why would it make it conditional if it's happening within the limits of what dbus supports?

Reply:
> It's a crate-level attribute.

Ughh.. :(

> Why would it make it conditional if it's happening within the limits of what dbus supports?

Because we don't need it at all. The whole reason the test is failing is because we don't allow this level of recursion.

But maybe it's a good thing that we limit recursion at compile-time? We can then remove the checks at serializer end and only check and test the deserialization? Or maybe there is no need for checks at all then as compiler knows the type being deserialized to? Need to look more closely into what this new compiler check is all about..

Reply:
I digged a bit: https://github.com/rust-lang/rust/issues/113160

Reply:
ah, so a regression in Rust. Even better. :)

Reply:
They don't seem to see it as a regression. We should have a solution in zbus. Bumping the default recursion limit shouldn't change anything.